src/main/java/org/nrnb/gsoc/enrichment/ui/EnrichmentTableModel.java

```diff
@@ -260,6 +260,96 @@ private boolean inSource(List<TermSource> sources, String termName) {
         return false;
     }
 
+    public void removeRedundantTerms(List<TermSource> sources, double cutoff) {
+        List<CyRow> rows = cyTable.getAllRows();
+        Long[] rowArray = new Long[rows.size()];
+        int i = 0;
+        for (CyRow row : rows) {
+            String termCategory = row.get(EnrichmentTerm.colSource, String.class);
+            if (sources.size() == 0 || inCategory(sources, termCategory)) {
+                rowArray[i] = row.get(EnrichmentTerm.colID, Long.class);
+                i++;
+            }
+        }
+
+        rowNames = removeRedundancy(rowArray, i, cutoff);
```

Comment:
Pass rows.size() here instead of i

Reply:
Please refer https://github.com/cytoscape/enrichment-table-app/pull/81#discussion_r916441243